functions/index.js

```diff
@@ -1,3 +1,120 @@
+
+// The Cloud Functions for Firebase SDK to create Cloud Functions and triggers.
+const {logger} = require("firebase-functions");
+const {onRequest} = require("firebase-functions/v2/https");
+const {onDocumentCreated} = require("firebase-functions/v2/firestore");
+const nodemailer = require('nodemailer');
+
+// The Firebase Admin SDK to access Firestore.
+const {initializeApp} = require("firebase-admin/app");
+const {getFirestore, Timestamp, arrayUnion} = require("firebase-admin/firestore");
+
+//Scheduale management
+const {onSchedule} = require("firebase-functions/v2/scheduler");
+
+const {defineString} = require('firebase-functions/params')
+
+
+initializeApp();
+
+const db = getFirestore();
+
+const email = defineString('FUNCTIONS_COMPANY_EMAIL')
+const password = defineString('FUNCTIONS_COMPANY_PASSWORD')
+
+const transporter = nodemailer.createTransport({
+  service: 'gmail', 
+  auth: {
+    user: email.value(), 
+    pass: password.value(), 
+  },
+});
+
+// this function send the email with a random 6-digit code
+exports.sendVerificationEmail = onRequest(async (req, res) => {
+
+  const email = req.body.data?.email;
+  const associationUid = req.body.data?.associationUid;
```

Comment:
You should strongly consider doing data-sanitization on the request parameters, as any malicious user could put anything here. For example, anyone could put their own email address and thus could claim any association. This is a major security issue that should be addressed as soon as possible.